platform/machine/do/cluster.go

```diff
@@ -147,3 +157,13 @@ func (dc *cluster) Destroy() {
 
 	dc.BaseCluster.Destroy()
 }
+
+// shouldRetry returns if the error is from DigitalOcean and we should
+// retry the request which generated it
+func shouldRetry(err error) bool {
+	errResp, ok := err.(*godo.ErrorResponse)
+	if !ok {
+		return false
+	}
+	return errResp.Response.StatusCode == 422
```

Comment:
You typically want to retry on 5xx errors; want to add those in too while we're at it?

Reply:
Sure, done.